deisctl/cmd/upgrade.go

```diff
@@ -0,0 +1,110 @@
+package cmd
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/deis/deis/deisctl/backend"
+	"github.com/deis/deis/deisctl/etcdclient"
+)
+
+// UpgradePrep stops and uninstalls all components except router and publisher
+func UpgradePrep(b backend.Backend) error {
+	var wg sync.WaitGroup
+
+	b.Stop([]string{"database", "registry@*", "controller", "builder", "logger", "logspout"}, &wg, Stdout, Stderr)
+	wg.Wait()
+	b.Destroy([]string{"database", "registry@*", "controller", "builder", "logger", "logspout"}, &wg, Stdout, Stderr)
+	wg.Wait()
+
+	b.Stop([]string{"store-volume", "store-gateway@*"}, &wg, Stdout, Stderr)
+	wg.Wait()
+	b.Destroy([]string{"store-volume", "store-gateway@*"}, &wg, Stdout, Stderr)
+	wg.Wait()
+
+	b.Stop([]string{"store-metadata"}, &wg, Stdout, Stderr)
+	wg.Wait()
+	b.Destroy([]string{"store-metadata"}, &wg, Stdout, Stderr)
+	wg.Wait()
+
+	b.Stop([]string{"store-daemon"}, &wg, Stdout, Stderr)
+	wg.Wait()
+	b.Destroy([]string{"store-daemon"}, &wg, Stdout, Stderr)
+	wg.Wait()
+
+	b.Stop([]string{"store-monitor"}, &wg, Stdout, Stderr)
+	wg.Wait()
+	b.Destroy([]string{"store-monitor"}, &wg, Stdout, Stderr)
+	wg.Wait()
+
+	fmt.Fprintln(Stdout, "The platform has been stopped, but applications are still serving traffic as normal.")
+	fmt.Fprintln(Stdout, "Your cluster is now ready for upgrade. Install a new deisctl version and run deisctl take-over.")
```

Comment:
@iancoffey Doesn't match the actual command that does the take-over.  Should be `upgrade-takeover`.

Also, for maximum clarity and consistency, you might consider using backticks to demarcate the boundaries of the command you're asking them to run.  That's done elsewhere in deisctl.  As a for instance, this is the message provided after a `deisctl install platform` reads as follows:

```
Please run `deisctl start platform` to boot up Deis.
```